lib/models/room.js

```diff
@@ -56,18 +72,24 @@ function synthesizeReceipt(userId, event, receiptType) {
  * when the call to <code>sendEvent</code> was made. If "<b>end</b>", pending messages
  * will always appear at the end of the timeline (multiple pending messages will be sorted
  * chronologically). Default: "chronological".
+ * @param {boolean} [opts.timelineSupport = false] Set to true to enable improved
+ * timeline support.
+ *
  * @prop {string} roomId The ID of this room.
  * @prop {string} name The human-readable display name for this room.
- * @prop {Array<MatrixEvent>} timeline The ordered list of message events for
- * this room, with the oldest event at index 0.
+ * @prop {Array<MatrixEvent>} timeline The live event timeline for this room,
+ * with the oldest event at index 0. Present for backwards compatibility -
+ * prefer getLiveTimeline().getEvents().
```

Comment:
You can do *horrible* things to migrate people over for the record.

```javascript
function Room() { // constructor
  var self = this;
  Object.defineProperty(this, 'timeline', {
    get: function () {
      return self.getLiveTimeline().getEvents();
    }
  });
}
```

Any time people refer to `room.timeline` the `get` function will be invoked. If this provides equivalent backwards-compat, I'm happy to nuke the `timelineSupport` gate flag.